folsom-opentelemetry-metrics/src/main/java/com/spotify/folsom/client/OpenTelemetryMetrics.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2014-2015 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.spotify.folsom.client;
+
+import com.spotify.folsom.GetResult;
+import com.spotify.folsom.MemcacheStatus;
+import com.spotify.folsom.Metrics;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.DoubleHistogram;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.ObservableDoubleMeasurement;
+
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+public class OpenTelemetryMetrics implements Metrics {
+
+    private static final long ONE = 1L;
+
+    private static final Attributes HIT_TAGS = tag("result", "hit");
+    private static final Attributes MISS_TAGS = tag("result", "miss");
+    private static final Attributes FAILURE_TAGS = tag("result", "failure");
+    private static final Attributes SUCCESS_TAGS = tag("result", "success");
+
+    private final DoubleHistogram durations;
+    private final LongCounter gets;
+    private final LongCounter multigets;
+    private final LongCounter deletes;
+    private final LongCounter sets;
+    private final LongCounter incrDecrs;
+    private final LongCounter touches;
+
+    private final Set<OutstandingRequestsGauge> outstandingRequests = new CopyOnWriteArraySet<>();
+
+    /**
+     * Constructs an instance of {@link com.spotify.folsom.client.OpenTelemetryMetrics},
+     * initializing various metrics and gauges for monitoring Memcache-related activities using OpenTelemetry.
+     *
+     * @param meter The {@link io.opentelemetry.api.metrics.Meter} instance used to create and register metrics and gauges.
+     *              Metrics such as histograms and counters will be associated with specific Memcache
+     *              request types, while gauges will provide real-time observability for outstanding
+     *              requests and global connections.
+     */
+    public OpenTelemetryMetrics(final Meter meter) {
```

Comment:
Nit: usually, all the OTel instrumentations receive an `OpenTelemetry` instance and retrieve `Meter`/`Tracer` themselves. That is because usually the instrumentations themselves set the instrumentation scope name to something that identifies what produced the telemetry. E.g.:
```java
public OpenTelemetryMetrics(final OpenTelemetry openTelemetry) {
  Meter meter = openTelemetry.getMeter("folsom"); // or "com.spotify.folsom"
  // ...
```